dffbackend/Filters/ApiKeyAttribute.cs

```diff
@@ -0,0 +1,63 @@
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using dffbackend.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace dffbackend.Filters;
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+public class ApiKeyAttribute : Attribute, IAuthorizationFilter
+{
+    private const string API_KEY_HEADER_NAME = "X-Factor-API-Key";
+
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        var submittedApiKey = GetSubmittedApiKey(context.HttpContext);
+
+        // var apiKey = GetApiKey(context.HttpContext);
+
+        // TODO: apply the commented code below in a following manner:
+        // From configuration (env variable) fetch encryption/decryption password
+        // Encrypt key and check against DB
+        // Use IsApiKeyValid method to validate api key and replace the IsApiKeyValidInDb call below
+
+        if (!IsApiKeyValidInDb(submittedApiKey, context.HttpContext))
```

Comment:
We also need a way to get the identity of the factor here for future storage of the proof. Can be done in a separate issue.